compiler/rustc_resolve/src/late.rs

```diff
@@ -693,6 +700,9 @@ struct LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
 
     /// Count the number of places a lifetime is used.
     lifetime_uses: FxHashMap<LocalDefId, LifetimeUseSet>,
+
+    /// Record poly-trait-ref, only used for diagnostic.
+    with_poly_trait_ref: FxHashMap<NodeId, RedundantPolyTraitRef>,
```

Comment:
More concretely, I'm wondering if instead of an extra field would it be possible to do something similar to https://github.com/rust-lang/rust/blob/b3bd7058c139e71bae0862ef8f8ac936208873e9/compiler/rustc_resolve/src/late/diagnostics.rs#L3161-L3163 where all the logic happens inside the `callback`. It might not be possible to do :thinking:

Reply:
Done. Thank you. I find self.diag_metadata.current_where_predicate has this information.